src/build/mod.rs

```diff
@@ -364,17 +365,31 @@ impl BuildQueue {
                 }
             }
 
+            let pbh = build.pbh;
+
+            // window/progress notification that we are about to build
+            pbh.notifier.notify_begin();
```

Comment:
As I said above, I guess it's not wonderful the notifier trait impl hangs off `pbh` for progress updates...